### DISCUSSION TITLE
GPIO Button working occasionally

### DISCUSSION THREAD

**Discussion by gitfoehn** [photobooth-app/photobooth-app#124] [Q&A] [answered]
Hey, 

I just installed the latest version and moved all the hardware in an actuall box. Since then I am having trouble with my GPIO Button. 
They allways work right after starting the Photobooth. However, ofter after the first picture is taken via Button Click, it seems like it is not recognizing my button press anymore. 
While testing there also where 2-3 times where I was able to trigger via button 5-10 times before the button stops working. Even when the GPIO buttons are not working I could still trigger a picture via clicking the  "take picture button" in the UI.

I am so confussed, because exept the actuall box it is the exact setup I was using the entire time and I never had issues with that. 

Here come some basic Infos about my configuration:

_RaspberryPi 4 8gb 
Raspberry Pi OS Bookworm 
Pi Camera 3 + Canon Eos 200D
The UI runs directly on the pi and is shown via kiosk mode on an old laptop monitor via HDMI._

I've used those buttons (all have them have worked before) but seem to struggle now.

https://www.berrybase.de/en/massive-arcade-button-100mm-beleuchtet-led-12v-dc?number=AB100L12-R&c=81
_Connected to Com and No_ 

https://www.ebay.de/itm/225721194021?hash=item348e081625:g:rigAAOSwB45lArUb&amdata=enc%3AAQAIAAABALI9zA3nYDZ7QJW8yDuGRGDjXSEaAgPsu4A%2FXtDRRbLBS77PCWrdz7PRqPDtexz%2FabftgUelBNOgbweytGm3Y1MdPb71%2BJaVhyu%2Fz%2Feye8%2B9vmFxHwrcL9ewOxUhBQE6bJHZ9MrFtfagiwb8Ago7dPQHNWDATlDXKotBfjkc81qw8%2B2hRbC1pXOAjyO9TDzOQzvtgNA%2BYw4ITN5oMJaWR4OqNvsqFfXb6Ni8tZc5xR%2FaRAiVlkgHzKjbE9HKV6dwZ3oNsBCli89inFixGmf6XkWA%2BW07gN4IxMx9PKCkWaVA1C1qu8adW8ql7Q2Az5AaZzb0F%2Bp1vLDTuQZFdgOmnrU%3D%7Ctkp%3ABk9SR5S8gdb3Yg 
_Connected to Com and No_ 

And a standard button like here:
https://www.roboter-bausatz.de/p/50-miniatur-drucktaster-tastschalter-6x6x4-3mm?sPartner=8&gclid=CjwKCAiAxreqBhAxEiwAfGfndNIjhMqzA-VvmIh51rWokJCS1C-2BczJjPXPdcl9XH-epPIJrRE7jxoCTsAQAvD_BwE

I connected all those buttons to the dafault gpio pin(27) and ground. I also tried another gpio (17) but had the same issue. 

Maybe someone has an idea what I could try. 

Thanks and cheers,
foehn

**Comment by mgineer85** [comment]
Let me try similar on hardware. Software wise it’s tested automatically, so I do not expect an issue there.

**Reply by gitfoehn** [reply]
Thanks @mgrl,

in the meantime I did another fresh install on another sd card. 
I went from pin 27 to 17 to 26. On all of them it worked first but after a cupple restarts (due to changing settings) I discovered the issues mentioned befor. 
I am currently on pin 16 and so far it works. I hope it makes it through the night!

**Reply by mgineer85** [reply]
Hi @gitfoehn , thank you for reporting.
I was able to reproduce this error and found it is an issue with the underlying library GPIOzero.
Unfortunately it's long time an issue and never finally been fixed.

I changed the code to work around that and release a new version in a few moments.

**Reply by mgineer85** [reply]
https://github.com/photobooth-app/photobooth-app/releases/tag/v0.12.1

**Reply by gitfoehn** [reply]
@mgrl, thanks for the investigation and the fix.

**Reply by peterwup** [reply]
You wrote "Software-debouncing is disabled", is it possible that lead to the following issue: 

Pressing on the "Take picture" button on the screen: 
- Countdown
- Smiley
- white screen
- taken picture is shown

Pressing short on a hardware button which is connected to Gpio Pin Take1Pic = 27
- Countdown
- Smiley
- White screen
- 9 animated black dots
- Smiley
- taken picture is shown
- 3 picture in the gallery

**Reply by mgineer85** [reply]
No this cannot happen. Three images you would see if there is a collage.

**Reply by mgineer85** [reply]
Could you post logs when you reproduce the issue? When you press f5 to refresh you still see 3 images?

**Reply by peterwup** [reply, marked as answer]
Sorry, my fault. I have to improve my soldering skills. 
I found the collage in the log too and checkd at the end the complete hardware.
Now it works as expected.

**Reply by mgineer85** [reply]
Hehe good it’s solved 🥳
